### PR TITLE
lib: fixed compilation error when USE_UNIX_SOCKETS is not defined

### PR DESCRIPTION
**Pull request by b5i**
Fix for #13620.

### CODE REVIEW

File: lib/socketpair.h [hunk 1]
```diff
@@ -38,7 +38,14 @@
 #define wakeup_write     swrite
 #define wakeup_read      sread
 #define wakeup_close     sclose
-#define wakeup_create(p) Curl_socketpair(AF_UNIX, SOCK_STREAM, 0, p)
+
+#ifdef USE_UNIX_SOCKETS
+#define CURL_SOCKET_FAMILY AF_UNIX
+#else
+#define CURL_SOCKET_FAMILY AF_INET
+#endif
```
Comment (bagder) [inline lib/socketpair.h:46]: If the native `socketpair()` function is used, it cannot be changed to AF_INET. And if the non-native implementation is used, the family is not used so might as well just be set to zero.

Also, no need to use `CURL_` prefix for a private define.

May I suggest:

```suggestion
#if defined(USE_UNIX_SOCKETS) && defined(HAVE_SOCKETPAIR)
#define SOCKET_FAMILY AF_UNIX
#elif !defined(USE_UNIX_SOCKETS) && !defined(HAVE_SOCKETPAIR)
#define SOCKET_FAMILY 0 /* not used */
#else
#error "unsupported unix domain and socketpair build combo"
#endif
```

File: lib/socketpair.h [hunk 2]
```diff
@@ -38,7 +38,14 @@
...
+
+#define wakeup_create(p) Curl_socketpair(CURL_SOCKET_FAMILY, SOCK_STREAM, 0, p)
```
Comment (bagder) [inline lib/socketpair.h:48]:
```suggestion
#define wakeup_create(p) Curl_socketpair(SOCKET_FAMILY, SOCK_STREAM, 0, p)
```

**Review by bagder — CHANGES_REQUESTED** [review]

File: lib/socketpair.h (hunk 1 above)
Comment (b5i) [reply lib/socketpair.h:46]: Thanks! I didn't know about that

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**